### PR TITLE
Super parser intro

### PR DESCRIPTION
**Pull request by kgromadzki**

### CODE REVIEW

File: whylog/config/super_parser.py [hunk 1]
```diff
@@ -0,0 +1,62 @@
...
+
+    def get_ordered_group(self, line):
+        match = self.regex.match(line)
+        if not match:
+            return self.NO_MATCH
+        groups = match.groups()
+        result = []
+        for group_nr in self.group_order:
+            conv_type = self.convertions.get(group_nr)
+            to_convert = groups[group_nr - 1]
```
Comment (konefalg) [inline whylog/config/super_parser.py]: `group_to_convert`?

File: whylog/config/super_parser.py [hunk 2]
```diff
@@ -0,0 +1,62 @@
...
+        return self.serialize() == other.serialize()
+
+    def get_ordered_group(self, line):
+        match = self.regex.match(line)
+        if not match:
+            return self.NO_MATCH
+        groups = match.groups()
+        result = []
+        for group_nr in self.group_order:
+            conv_type = self.convertions.get(group_nr)
```
Comment (konefalg) [inline whylog/config/super_parser.py]: `convertion_type`?

File: whylog/config/super_parser.py [hunk 3]
```diff
@@ -0,0 +1,62 @@
...
+        return {
+            'regex_str': self.regex.pattern,
+            'group_order': self.group_order,
+            'convertions': self.convertions
+        }
+
+    def __eq__(self, other):
+        return self.serialize() == other.serialize()
+
+    def get_ordered_group(self, line):
```
Comment (konefalg) [inline whylog/config/super_parser.py]: This function returns list, so `get_ordered_groups` would be better name.

File: whylog/config/super_parser.py [hunk 4]
```diff
@@ -0,0 +1,62 @@
...
+
+@six.add_metaclass(ABCMeta)
+class AbstractSuperParser(object):
+    @abstractmethod
+    def get_ordered_groups(self, line):
+        pass
+
+
+class RegexSuperParser(AbstractSuperParser):
+    NO_MATCH = []
```
Comment (ppolewicz) [inline whylog/config/super_parser.py]: `tuple()` would he better - nobody will append nothing to it, can use less memory etc

File: whylog/config/super_parser.py [hunk 5]
```diff
@@ -0,0 +1,62 @@
...
+
+import six
+
+from whylog.converters import CONVERTION_MAPPING, STRING
+
+
+@six.add_metaclass(ABCMeta)
+class AbstractSuperParser(object):
+    @abstractmethod
+    def get_ordered_groups(self, line):
```
Comment (ppolewicz) [inline whylog/config/super_parser.py:12]: Missing important docstring

File: whylog/config/super_parser.py [hunk 6]
```diff
@@ -0,0 +1,62 @@
...
+            return self.NO_MATCH
+        groups = match.groups()
+        result = []
+        for group_nr in self.group_order:
+            convertion_type = self.convertions.get(group_nr)
+            group_to_convert = groups[group_nr - 1]
+            if convertion_type is None:
+                result.append((STRING, group_to_convert))
+                continue
+            converter = CONVERTION_MAPPING[convertion_type]
```
Comment (ppolewicz) [inline whylog/config/super_parser.py:62]: This should be a converter instance and it looks like the object is being reused

Comment (kgromadzki) [inline whylog/config/super_parser.py:62]: converter isn't an object. Is a class name. I just invoke classmethods

Comment (ppolewicz) [inline whylog/config/super_parser.py:62]: yes, this is exactly a problem. You need to use a converter object which you need to store somewhere..

**Pull request merged**